### ISSUE TITLE
[ColorCalibration] On the grey I expect monochrome output but at one point it gives a colored output

### ISSUE THREAD

**Issue by chhil** [darktable-org/darktable#6854]
darktable-3.3.0+1487~gf4bdf79ea-win64

![image](https://user-images.githubusercontent.com/1119439/98784882-c73b3900-2421-11eb-8114-8e1d1c988cc0.png)

Red and green were set and was moving the blue to see what the output looks like, at one point it loses its monochrome output.

I wonder since the net some of values of the sliders is 0, it removes any impact.
p.s. I don't know how to use this module, just playing around with the new toy.

**Comment by matt-maguire** [comment]
If you add up the three coefficients, they come to zero, which causes the grey conversion to disengage. Probably it should be checking that each individual slider is zero rather than checking if their sum is zero...

**Comment by chhil** [comment]
Channel mixer behavior leaves it grey. I don't know what behavior to expect here.

**Comment by matt-maguire** [comment]
Maybe @aurelienpierre can comment on whether there is a good reason or not for this new behaviour.

**Comment by aurelienpierre** [comment]
It's a bug.

**Comment by aurelienpierre** [comment]
Well, let me rephrase that. That's a bug BUT your settings are really insane. You are not supposed to use negative coeffs that large. And also, the fact that the sum of your settings = 0 means the normalization will divide by zero, so the program bypasses the grey to avoid that.

**Comment by chhil** [comment]
I am was just playing with the sliders to get a fee for the new module.

If such negative coeffs are not to be used, possibly warn the user? Normalize should handle that right?
Should this new one mimic the grey in the old channel mixer?

I have questions because I don't know what to expect.

**Comment by aurelienpierre** [comment]
Normalize handles that right. It divides each R, G, B parameter by (R + G + B) to preserve the "luminance". Negative coeffs can be used with caution, as long as the total sum of params ≠ 0. If they are = 0, the division would result in NaN and it is avoided in the soft, which results in a bypass of grey conversion.

The old channel mixer did not normalize anything, so it did not have this problem.

The problem here is user responsibility. Defining negative coeffs is a valid use, although it is playing with fire. If I prevent that, I reduce the possibilities of the module for educated users. If I don't, I let uneducated users burn themselves. 

Adding a warning is easy and safe, without degrading the abilities of the module.

**Comment by junkyardsparkle** [comment]
> Defining negative coeffs is a valid use, although it is playing with fire. If I prevent that, I reduce the possibilities of the module for educated users. If I don't, I let uneducated users burn themselves

Use soft limits on the sliders?

**Comment by Mark-64** [comment]
Negative values don't need to be large to produce the bug

![immagine](https://user-images.githubusercontent.com/43290988/98985343-b70b8280-2523-11eb-9e6c-1a075b6b615f.png)

**Comment by aurelienpierre** [comment]
negative values don't need to be large to be R + G + B = 0.

Now, in real-life use cases, these parameters make no sense.

**Comment by rawfiner** [comment]
Would it be possible to replace the norm by max(norm, 0.01) or something like that when one or several coefs are negative to avoid the issue?

**Comment by matt-maguire** [comment]
I don't understand why this is an issue to be avoided. You clearly want this module to be inactive when all the sliders are zero. Setting the sliders in such a way that they add up to zero is not reasonable unless all are set to zero.

Note also that the norm can be negative in cases where the total of the sliders is negative, so forcing it to 0.01 in case cases will not have a desirable effect.
